modules/new_serverpod_auth/serverpod_auth_session/serverpod_auth_session_server/lib/src/business/user_sessions.dart

```diff
@@ -0,0 +1,181 @@
+import 'dart:io';
+
+import 'package:serverpod/protocol.dart';
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_auth_session_server/serverpod_auth_session_server.dart';
+import 'package:serverpod_auth_session_server/src/util/random_string.dart';
+import 'package:serverpod_auth_session_server/src/util/salted_hash.dart';
+import 'package:serverpod_auth_user_server/serverpod_auth_user_server.dart';
+
+/// Management functions for [UserSession]s.
+///
+/// This should be used instead of [UserSession.db].
+abstract final class UserSessions {
+  /// Looks up the `AuthenticationInfo` belonging to the [key].
+  ///
+  /// The [key] maps to the [UserSession.sessionKeyHash] field.
+  ///
+  /// In case the authentication could not be resolved, this returns `null` and does never throw.
+  static Future<AuthenticationInfo?> authenticationHandler(
+    final Session session,
+    final String key,
+  ) async {
+    try {
+      final secret = key;
+      final signInSalt =
```

Comment:
To be discussed: Do we need to retain the "salting" of the session key?

I would think having a strong random key should be enough, and if the developer wants to purge all sessions, they can just drop the table (instead of changing the salt in a config).

Reply:
Moved the pattern to use a `pepper` (static configured password) to match the email accounts.

The email accounts also use the `email` as the salt. We could also add a per-session salt if desired, just so that a leaking of the DB + configuration does not result in the attacker being (easily) able to generate the full key to make requests. Though in that case we might just check that the session key length is sufficient.